### ISSUE TITLE
Error: dependency on a module not visible from current build project: 990507d3-3527-4c54-bfe9-0ca3c9c6247a(com.dslfoundry.plaintextgen)

### ISSUE THREAD

**Issue by gerhardzach** [JetBrains/MPS-extensions#318]
Using MPS 2020.1.4 and the com.dslfoundry.plaintextgen plugin 2020.1.4 I created a language and a sandbox solution (works as expected). When trying to create a Build-Solution (for Standalone IDE) using the wizard I always run into "Error: dependency on a module not visible from current build project: 990507d3-3527-4c54-bfe9-0ca3c9c6247a(com.dslfoundry.plaintextgen)"
So far I've found no documentation on how to solve this problem. (https://dslfoundry.com/plaintextgen-tutorial/ unfortunately also doesn't show what to consider when creating a Build-Solution for a language that uses plaintextgen.)

**Comment by sergej-koscejev** [comment]
This is not a bug in MPS-extensions. You need to tell the build script where to find the library. I have a post with explanations and step-by-step instructions here: https://specificlanguages.com/posts/build-script-errors/dependency-on-a-module-not-visible-from-current-build-project/

**Comment by gerhardzach** [comment]
Thanks for the link!
But when I click the build-script link for 'de.itemis.mps.extensions' in your instruction then I get the following error-message in the browser:
MPS - HTTP Support Plugin
The requested node has not been found
Node reference: r:fb24ac52-5985-4947-bba9-25be6fd32c1a(de.itemis.mps.extensions.build)/3410197114990977083
Project: Test_Project

Also Ctrl+M search for de.itemis.mps.extensions.build shows no result.

I'm a bit confused right now. 
Do I maybe need to use the source version of the mps-extensions for that? because currently I use the installed version of the com.dslfoundry.plaintextgen plugin (via 'Settings->Plugins')?

**Comment by sergej-koscejev** [comment]
Yes, you need a packaged version (from the [itemis Nexus](https://projects.itemis.de/nexus/content/repositories/mbeddr/de/itemis/mps/extensions/)) because
1) apparently the build solution isn't available from the plugin repository, and
2) when building from command line you probably want to download a particular version anyway and use it as a project library rather than MPS plugin, to make version management and upgrades easier.

**Comment by gerhardzach** [comment]
Thanks a lot! That was the 'missing link' - got it working now for a small example.

I think it would be helpful if a short note on that would be available here: https://jetbrains.github.io/MPS-extensions/extensions/plaintext-gen/ - so that people like me who follow the Quickstart section are aware that they may run into this kind of trouble when using the plugin instead of a packaged version. 
Otherwise the issue can be closed I guess.

**Comment by eugenschindler** [comment]
@gerhardzach thanks for reporting and for the follow-ups.
@mvlassiev could it be that the automatic deploy of plugins to the marketplace automatically strips out the build solution (which is always needed if you want to use plaintextgen plugin in builds)?

**Comment by eugenschindler** [comment]
@mvlassiev did you have a chance to think about my proposals/thoughts?
